ui/src/alerting/actions/checks.ts

```diff
@@ -140,50 +141,59 @@ export const saveCheckFromTimeMachine = () => async (
   dispatch: Dispatch<any>,
   getState: GetState
 ) => {
-  const state = getState()
-  const {
-    orgs: {
-      org: {id: orgID},
-    },
-  } = state
+  try {
+    const state = getState()
+    const {
+      orgs: {
+        org: {id: orgID},
+      },
+    } = state
 
-  const {
-    draftQueries,
-    alerting: {check},
-  } = getActiveTimeMachine(state)
-
-  const labels = get(check, 'labels', []) as Label[]
-  const checkWithOrg = {
-    ...check,
-    query: draftQueries[0],
-    orgID,
-    labels: labels.map(l => l.id),
-  } as PostCheck
-
-  const resp = check.id
-    ? await api.patchCheck({checkID: check.id, data: checkWithOrg})
-    : await api.postCheck({data: checkWithOrg})
-
-  if (resp.status === 201 || resp.status === 200) {
-    dispatch(setCheck(resp.data))
-    dispatch(checkChecksLimits())
-  } else {
-    throw new Error(resp.data.message)
+    const {
+      draftQueries,
+      alerting: {check},
+    } = getActiveTimeMachine(state)
+
+    const labels = get(check, 'labels', []) as Label[]
+
+    const checkWithOrg = {
+      ...check,
+      query: draftQueries[0],
+      orgID,
+      labels: labels.map(l => l.id),
+    } as PostCheck
+
+    const resp = check.id
+      ? await updateCheckFromTimeMachine(checkWithOrg)
+      : await api.postCheck({data: checkWithOrg})
+
+    if (resp.status === 200 || resp.status === 201) {
+      dispatch(setCheck(resp.data))
+      dispatch(checkChecksLimits())
+
+      dispatch(push(`/orgs/${orgID}/alerting`))
+      dispatch(setTimeMachineCheck(RemoteDataState.NotStarted, null))
+    } else {
+      throw new Error(resp.data.message)
+    }
+  } catch (e) {
+    console.error(e)
+    dispatch(notify(copy.createCheckFailed(e.message)))
   }
 }
 
-export const updateCheck = (check: Partial<Check>) => async (
-  dispatch: Dispatch<Action | NotificationAction>
-) => {
-  const resp = await api.putCheck({checkID: check.id, data: check as Check})
+const updateCheckFromTimeMachine = async (check: Check) => {
+  // todo: refactor after https://github.com/influxdata/influxdb/issues/16317
+  const getCheckResponse = await api.getCheck({checkID: check.id})
 
-  if (resp.status === 200) {
-    dispatch(setCheck(resp.data))
-  } else {
-    throw new Error(resp.data.message)
+  if (getCheckResponse.status !== 200) {
+    throw new Error(getCheckResponse.data.message)
   }
 
-  dispatch(setCheck(resp.data))
+  return api.putCheck({
+    checkID: check.id,
+    data: {...check, ownerID: getCheckResponse.data.ownerID},
+  })
```

Comment:
does this need to dispatch a redux action or is this all good? looks like formerly it was `setCheck` ?

Reply:
it is happening on line 171 inside of `saveCheckFromTimeMachine`- the diff is misleading since I've replaced the `updateCheck` function that was being called directly from the component with `updateCheckFromTimeMachine` which is being called from `saveCheckFromTimeMachine` to hide the complexity with the get check we have to do because of the weirdness with putCheck requiring an ownerID.